AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,72 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+
+  // Mark the node for layout in the next layout pass
+  [self invalidateCalculatedLayout];
+  
+  // Hook for subclasses to get size invalidation changes
+  [self didInvalidateSize];
+  
+  if (_supernode) {
+    ASDisplayNode *supernode = _supernode;
+    __instanceLock__.unlock();
+    // Cause supernode's layout to be invalidated
+    // We need to release the lock to prevent a deadlock
+    [supernode invalidateSize];
+    return;
+  }
+  
+  // We are now at the root node trigger a layout pass
+  [self setNeedsLayout];
+  
+  __instanceLock__.unlock();
+}
+
+- (void)didInvalidateSize
+{
+    
+}
+
+- (void)sizeToFit
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+  
+  [self setNeedsLayout];
+  
+  CGSize maxSize = _supernode ? _supernode.bounds.size : CGSizeMake(CGFLOAT_MAX, CGFLOAT_MAX);
+  CGSize newSize = [self sizeThatFits:maxSize];
+  
+  CGRect oldBounds = self.bounds;
+  CGSize oldSize = oldBounds.size;
+  
+  if (! CGSizeEqualToSize(oldSize, newSize)) {
+    self.bounds = (CGRect){ oldBounds.origin, newSize };
+    
+    // Frame's origin must be preserved. Since it is computed from bounds size, anchorPoint
+    // and position (see frame setter in ASDisplayNode+UIViewBridge), position needs to be adjusted.
+    CGPoint anchorPoint = self.anchorPoint;
+    CGPoint oldPosition = self.position;
+    CGFloat xDelta = (newSize.width - oldSize.width) * anchorPoint.x;
+    CGFloat yDelta = (newSize.height - oldSize.height) * anchorPoint.y;
+    self.position = CGPointMake(oldPosition.x + xDelta, oldPosition.y + yDelta);
+  }
+  
+  __instanceLock__.unlock();
+}
+
+- (CGSize)sizeThatFits:(CGSize)size
```

Comment:
Same for this method - seems like we should consider having it in the future, and that we should implement this in _ASDisplayView, but for now this operation is easy enough for developers to perform with the existing layoutThatFits: API?